src/Api/Controller/AbstractSerializeController.php

```diff
@@ -197,6 +226,106 @@ protected function buildParameters(ServerRequestInterface $request)
         return new Parameters($request->getQueryParams());
     }
 
+    /**
+     * Set the serializer that will serialize data for the endpoint.
+     *
+     * @param string $serializer
+     */
+    public function setSerializer($serializer)
```

Comment:
Perhaps all of these convenience methods should be added as extender methods, with an additional parameter being a callback that is evaluated to check whether the extension in question should be applied?

Reply:
Callback would have to be optional I believe
```php
->setSerializer($serializerClass, $optionalCallback)
->addInclude('relation' | [...relations], $optionalCallback)
->removeInclude('relation' | [...relations], $optionalCallback)
->addOptionalInclude('relation' | [...relations], $optionalCallback)
...
```

Reply:
What is `'relation' | [...relations]` here? Wouldn't it just be one (string) value like it is in the current event signatures?

Other than that, this makes sense! I think we should also have a `mutate` method here, and package all these up under that, like with ApiSerializer.

Reply:
well, the signature is inconsistent, for instance, addInclude in the event allows either a string or an array of strings, while addOptionalInclude allows only a string, I would rather make them consistent and have them both allow `string|array`, in the controller and the extender.

> Other than that, this makes sense! I think we should also have a mutate method here, and package all these up under that, like with ApiSerializer.

mutate ? we have prepareDataQuery and prepareDataForSerialization, so I'm not sure where `mutate` would come in here?

Reply:
> well, the signature is inconsistent, for instance, addInclude in the event allows either a string or an array of strings, while addOptionalInclude allows only a string, I would rather make them consistent and have them both allow `string|array`, in the controller and the extender.

Ah got it, makes sense to me.

> mutate ? we have prepareDataQuery and prepareDataForSerialization, so I'm not sure where `mutate` would come in here?

Sorry I meant the general idea of a callback that's applied, it's covered by those.